unstructured_ingest/v2/processes/connectors/delta_table.py

```diff
@@ -94,14 +94,17 @@ class DeltaTableUploader(Uploader):
     connection_config: DeltaTableConnectionConfig
     connector_type: str = CONNECTOR_TYPE
 
-    @requires_dependencies(["s3fs", "fsspec"], extras="s3")
+    @requires_dependencies(["s3fs", "fsspec", "boto3"], extras="s3")
     def precheck(self):
         secrets = self.connection_config.access_config.get_secret_value()
         if (
             self.connection_config.aws_region
             and secrets.aws_access_key_id
             and secrets.aws_secret_access_key
         ):
+            from urllib.parse import urlparse
+
+            from boto3 import client
```

Comment:
To the best of my knowledge, fsspec does not provide a function to check the region of an S3 bucket

Reply:
Also `fsspec` abstracts `s3fs`. If we know that's all we need, simply depend on that and use it rather than pulling in all of `fsspec` as a dependency. Assuming we can't use boto3 for all of the check.